lib/backend/etcdbk/etcd.go

```diff
@@ -217,9 +218,38 @@ func New(ctx context.Context, params backend.Params) (*EtcdBackend, error) {
 		watchDone:        make(chan struct{}),
 		buf:              buf,
 	}
+
 	if err = b.reconnect(); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	// Check that the etcd nodes are at least the minimum version supported
+	timeout, cancel := context.WithTimeout(ctx, time.Second*3*time.Duration(len(cfg.Nodes)))
+	defer cancel()
+	for _, n := range cfg.Nodes {
+		status, err := b.client.Status(timeout, n)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		vs := strings.Split(status.Version, ".")
+		var maj, min int
+		if len(vs) >= 2 {
+			if maj, err = strconv.Atoi(vs[0]); err != nil {
+				return nil, trace.Wrap(err)
+			}
+			if min, err = strconv.Atoi(vs[1]); err != nil {
+				return nil, trace.Wrap(err)
+			}
+		}
+
+		if maj < 3 || (maj == 3 && min < 3) {
+			return nil, trace.BadParameter("unsupported version of etcd %v for node %v, must be 3.3 or greater",
+				status.Version, n)
+		}
+
```

Comment:
Is there a reason you are not using github.com/coreos/go-semver/semver?

Reply:
On the slab engineering practices it said to limit external libraries, but it's probably best to use it here, I'll change it.

Reply:
I think as it's already used elsewhere in the codebase it's fine.